### PR TITLE
feat: Upgrade to latest foundry features

### PR DESCRIPTION
**Pull request by 0xrajath**
## Motivation:
Upgrade to latest foundry features

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -7,10 +7,16 @@ update   :; forge update
 install  :; forge install
 
 # Build & test
-build    :; forge clean && forge build --optimize --optimizer-runs 1000000
-test     :; forge clean && forge test -f ${RPC_URL} --fork-block-number ${BLOCK_NUMBER} --etherscan-api-key ${ETHERSCAN_API_KEY} $(call compute_test_verbosity,${V}) # Usage: make test [optional](V=<{1,2,3,4,5}>)
-match    :; forge clean && forge test -f ${RPC_URL} --fork-block-number ${BLOCK_NUMBER} --etherscan-api-key ${ETHERSCAN_API_KEY} -m ${MATCH} $(call compute_test_verbosity,${V}) # Usage: make match MATCH=<TEST_FUNCTION_NAME> [optional](V=<{1,2,3,4,5}>)
-report   :; forge clean && forge test -f ${RPC_URL} --fork-block-number ${BLOCK_NUMBER} --gas-report | sed -e/╭/\{ -e:1 -en\;b1 -e\} -ed | cat > .gas-report
+build    :; forge clean && forge build
```
Comment (AustinGreen) [inline Makefile:10]: Pretty sure forge clean happens automatically now

File: Makefile [hunk 2]
```diff
@@ -7,10 +7,16 @@ update   :; forge update
...
+test     :; forge clean && forge test --etherscan-api-key ${ETHERSCAN_API_KEY} $(call compute_test_verbosity,${V}) # Usage: make test [optional](V=<{1,2,3,4,5}>)
+match    :; forge clean && forge test --etherscan-api-key ${ETHERSCAN_API_KEY} -m ${MATCH} $(call compute_test_verbosity,${V}) # Usage: make match MATCH=<TEST_FUNCTION_NAME> [optional](V=<{1,2,3,4,5}>)
+report   :; forge clean && forge test --gas-report | sed -e/╭/\{ -e:1 -en\;b1 -e\} -ed | cat > .gas-report
+
+# Deploy and Verify Contract
+deploy-contract :; forge script script/DeployContract.s.sol:DeployContract --rpc-url ${RPC_MAINNET_URL} --broadcast --private-key ${PRIVATE_KEY} --verify --etherscan-api-key ${ETHERSCAN_API_KEY} -vvvv
+verify-payload :; forge script script/DeployContract.s.sol:DeployContract --rpc-url ${RPC_MAINNET_URL} --verify --etherscan-api-key ${ETHERSCAN_API_KEY} -vvvv
```
Comment (AustinGreen) [inline Makefile]: Let's line these up to improve readability

File: Makefile (hunk 1 above)
Comment (0xrajath) [reply Makefile:10]: Nope. Just tried with `forge build`. It doesn't remove the `/out` directory;.

File: Makefile (hunk 2 above)
Comment (0xrajath) [reply Makefile]: Done.

**Review by AustinGreen — APPROVED** [review]

**Pull request merged**